src/main/java/com/ldtteam/blockout/BOScreen.java

```diff
@@ -111,44 +114,49 @@ public BOScreen(final Window w)
         window = w;
     }
 
-    public static double getScale()
-    {
-        return scale;
-    }
-
-    private static void setScale(final Minecraft mc)
-    {
-        // Seems to work without the sides now
-        // Failsave
-        if (mc != null)
-        {
-            scale = mc.mainWindow.getGuiScaleFactor();
-        }
-    }
-
     @Override
+    @SuppressWarnings("deprecation")
     public void render(final MatrixStack ms, final int mx, final int my, final float f)
     {
         if (window.hasLightbox() && super.minecraft != null)
         {
             super.renderBackground(ms);
         }
 
-        setScale(minecraft);
-        final float oldZ = Minecraft.getInstance().getItemRenderer().zLevel;
-        Minecraft.getInstance().getItemRenderer().zLevel = MatrixUtils.getLastMatrixTranslateZ(ms);
+        final double fbHeight = minecraft.mainWindow.getFramebufferHeight();
+        final double fbWidth = minecraft.mainWindow.getFramebufferWidth();
+        final double guiWidth = Math.max(fbWidth, 320.0d);
+        final double guiHeight = Math.max(fbHeight, 240.0d);
 
-        ms.push();
-        ms.translate(x, y, 0);
-        window.draw(ms, mx - x, my - y);
-        ms.pop();
+        final float renderZlevel = MatrixUtils.getLastMatrixTranslateZ(ms);
+        final float oldZ = minecraft.getItemRenderer().zLevel;
+        minecraft.getItemRenderer().zLevel = renderZlevel;
```

Comment:
You need a none null check for minecraft in all this code.

Reply:
Then `if minecraft == null return` and we do no rendering

Reply:
You could add that too.